### ISSUE TITLE
User-specified loss function may not be reflected in compiled model

### ISSUE THREAD

**Issue by stsievert** [adriangb/scikeras#100]
As discussed in https://github.com/adriangb/scikeras/pull/88#discussion_r502738669, if the user specifies a loss function, it might get over-ridden by `build_fn`. That is, this is possible:

``` python
>>> wrapper = KerasRegressor(..., loss="mean_abs_error")
>>> wrapper._initialize()
# UserWarning: loss='mean_abs_error' but model compiled with 'mse'
>>> wrapper.model_.loss
"mse"
```

This is specifically the case when the default loss is changed; for KerasRegressor, the default is `loss=None`. When the user requests `loss="mean_abs_error"` but gets `loss="mse"`, I think an error should be raised.
